From ae9dc92202ce27a7e9d793666277c4dd47b681bc Mon Sep 17 00:00:00 2001
From: Rajvi Jingar <rajvi.jingar@intel.com>
Date: Fri, 10 Jan 2020 15:49:42 -0800
Subject: [PATCH 04/10] Added TGPIO pin check before input event read

This patch adds check for PTP_PINDESC_INPUTDISABLE flag before
performing read to input event.

Signed-off-by: Rajvi Jingar <rajvi.jingar@intel.com>
---
 tools/testing/selftests/ptp/testptp.c | 33 ++++++++++++++++++++-------------
 1 file changed, 20 insertions(+), 13 deletions(-)

diff --git a/tools/testing/selftests/ptp/testptp.c b/tools/testing/selftests/ptp/testptp.c
index 6ae2d14..a067336 100644
--- a/tools/testing/selftests/ptp/testptp.c
+++ b/tools/testing/selftests/ptp/testptp.c
@@ -370,20 +370,27 @@ int main(int argc, char *argv[])
 		} else {
 			puts("external time stamp request okay");
 		}
-		for (; extts; extts--) {
-			cnt = read(fd, &event, sizeof(event));
-			if (cnt != sizeof(event)) {
-				perror("read");
-				break;
-			}
-			printf("event index %u at %lld.%09u\n", event.index,
-			       event.t.sec, event.t.nsec);
-			fflush(stdout);
+		memset(&desc, 0, sizeof(desc));
+		desc.index = index;
+		if (ioctl(fd, PTP_PIN_GETFUNC2, &desc)) {
+			perror("PTP_PIN_GETFUNC2");
 		}
-		/* Disable the feature again. */
-		extts_request.flags = 0;
-		if (ioctl(fd, PTP_EXTTS_REQUEST, &extts_request)) {
-			perror("PTP_EXTTS_REQUEST");
+		if (!(desc.flags & PTP_PINDESC_INPUTDISABLE)) {
+			for (; extts; extts--) {
+				cnt = read(fd, &event, sizeof(event));
+				if (cnt != sizeof(event)) {
+					perror("read");
+					break;
+				}
+				printf("event index %u at %lld.%09u\n", event.index,
+				       event.t.sec, event.t.nsec);
+				fflush(stdout);
+			}
+			/* Disable the feature again. */
+			extts_request.flags = 0;
+			if (ioctl(fd, PTP_EXTTS_REQUEST, &extts_request)) {
+				perror("PTP_EXTTS_REQUEST");
+			}
 		}
 	}
 
-- 
2.7.4

